ospf6d/ospf6_main.c

```diff
@@ -173,6 +173,37 @@ FRR_DAEMON_INFO(ospf6d, OSPF6, .vty_port = OSPF6_VTY_PORT,
 		.n_yang_modules = array_size(ospf6d_yang_modules),
 );
 
+/* Max wait time for config to load before accepting hellos */
+#define OSPF6_PRE_CONFIG_MAX_WAIT_SECONDS 600
+
+static void ospf6_config_finish(struct event *t)
+{
+	zlog_err("OSPF6 configuration end timer expired after %d seconds.",
+		 OSPF6_PRE_CONFIG_MAX_WAIT_SECONDS);
+}
+
+static void ospf6_config_start(void)
+{
+	if (IS_OSPF6_DEBUG_EVENT)
+		zlog_debug("ospf6d config start received");
+	EVENT_OFF(t_ospf6_cfg);
+	event_add_timer(master, ospf6_config_finish, NULL,
+			OSPF6_PRE_CONFIG_MAX_WAIT_SECONDS, &t_ospf6_cfg);
+}
+
+static void ospf6_config_end(void)
+{
+	if (IS_OSPF6_DEBUG_EVENT)
+		zlog_debug("ospf6d config end received");
+	/* If the OSPFv3 config processing timer thread isn't running, then
+	 * we can safely assume that there is nothing more to do.
+	 */
+	if (!event_is_scheduled(t_ospf6_cfg))
```

Comment:
I think we can shoot directly EVENT_OFF() without checking if the thread is scheduled or not.

Reply:
If the config_start/config_end are always paired perfectly, it shouldn't be needed. However, if you look at the IS-IS and BGP config callbacks, you'll see that they turn off the event. It is more robust to turn it off unconditionally.  For example, from bgp_vty.c:

static void bgp_config_start(void)
{
#define BGP_PRE_CONFIG_MAX_WAIT_SECONDS 600
        EVENT_OFF(t_bgp_cfg);
        event_add_timer(bm->master, bgp_config_finish, NULL,
			BGP_PRE_CONFIG_MAX_WAIT_SECONDS, &t_bgp_cfg);
}